src/app/complex/complex-results/complex-results.component.ts

```diff
@@ -194,4 +221,40 @@ export class ComplexResultsComponent implements OnInit, AfterViewInit {
   set interactorTypeFilter(value: string[]) {
     this._interactorTypeFilter = value;
   }
+
+  public get allInteractorsInComplexSearch(): Interactor[] {
+    return this._allInteractorsInComplexSearch;
+  }
+
+  set allInteractorsInComplexSearch(value: Interactor[]) {
+    this._allInteractorsInComplexSearch = value;
+  }
+
+  setListView() {
+    this.DisplayType = this.LIST_VIEW;
+    this.reloadPage();
+  }
+
+  setComplexNavigatorView() {
+    this.DisplayType = this.COMPLEX_NAVIGATOR_VIEW;
+    this.reloadPage();
+  }
+
+  setFirstDisplayType(): void {
```

Comment:
`window.location.href` is not the proper way to redirect to the complex details page. Also, you're using localhost, which worked for testing locally, but it won't remote.

Try replacing it with
```
  setFirstDisplayType(): void {
    if (this._complexSearch.elements.length === 1) {
      const complexId = this._complexSearch.elements[0].complexAC;
      if (!!complexId) {
        // For some reason this is needed so the navigate call works
        this.router.routeReuseStrategy.shouldReuseRoute = function () {
          return false;
        };
        this.router.navigate(['/complex', complexId]);
      }
    }

    if (!this.DisplayType) {
      // Currently the list view is the default, as we are just launching the navigator view
      // Later on we can change the default view to be the list or navigator view based on number of results
      this.setListView();
    }
  }
```

I also changed the logic of the display type to have list be default for now, so when we first launch it we keep list as the default view. Maybe for the UX testing, we can have the navigator view as default, and then switch it back.